src/cascadia/TerminalApp/TerminalTab.h

```diff
@@ -41,7 +41,7 @@ namespace winrt::TerminalApp::implementation
                        std::shared_ptr<Pane> newPane);
 
         void ToggleSplitOrientation();
-        void UpdateIcon(const winrt::hstring iconPath);
+        void UpdateIcon(const winrt::hstring iconPath, const winrt::Microsoft::Terminal::Settings::Model::IconStyle& iconStyle);

```

Comment:
I agree with @KalleOlaviNiemitalo. `const&` is most useful when passing RAII types to avoid making costly copies. An integer is much much cheaper to pass by-value.

(It can also be useful under certain circumstances when passing >8 bytes on the Windows x64 ABI due to poor MSVC optimizations, but I'm sure those few cycles are fairly irrelevant in most circumstances.)

Reply:
You can hardly use the indication of other instances of this mistake as proof that it isn't a mistake. 😅 Otherwise what does this code imply?
https://github.com/microsoft/terminal/blob/9b70a40bf9efb7541981106e094009b79627ca86/src/types/ModifierKeyState.cpp#L100-L137

It turns a regular bitset into a `std::unordered_set<>` just to then test whether a given bit is set. It creates an entire hashmap instance just to do the equivalent of `a & b != 0`. Does this imply that binary operators are bad? (I've since replaced this particular code with `a & b != 0`.)

Passing primitive types by reference is a minuscule issue, so it's not really something I'm bothered by and I'll approve this PR regardless. But I believe if it came to a debate about this, that I could win it with fairly solid arguments.

Especially since statements like "it should not copy the value" are factually wrong. Passing parameters by value doesn't necessarily create a copy. Primitive types aren't classes after all.